ci/check-protobuf-stability.sh

```diff
@@ -35,42 +49,65 @@ function check_non_lf_protos() {
   readonly BUF_IMAGE_TMPDIR="$(mktemp -d)"
   trap 'rm -rf ${BUF_IMAGE_TMPDIR}' EXIT
 
-  # The `SYSTEM_PULLREQUEST_TARGETBRANCH` environment variable is defined by
-  # Azure Pipelines; in order to run this script locally, define it beforehand
-  # as the branch being targeted. For example:
-  #
-  #   SYSTEM_PULLREQUEST_TARGETBRANCH=main bash -x ci/check-protobuf-stability.sh
-  #
-  TARGET="${SYSTEM_PULLREQUEST_TARGETBRANCH:-main}"
-  echo "The target branch is '${TARGET}'."
+  echo "Checking protobufs against target '${BUF_TAG_TO_CHECK}'"
+  eval "$(dev-env/bin/dade assist)"
+  for buf_module in "${BUF_MODULES_AGAINST_STABLE[@]}"; do
+    # Starting with version 1.17 we split the default `buf.yaml` file into multiple config files
+    # This in turns requires that we pass the `--against-config` flag for any check that is run on versions > 1.17
+    if [[ $BUF_CONFIG_UPDATED ]]
+    then
+      buf breaking --config "${buf_module}" --against "$BUF_TAG_TO_CHECK" --against-config "${buf_module}"
+    else
+      buf breaking --config "${buf_module}" --against "$BUF_TAG_TO_CHECK"
+    fi
+  done
 
-  # For `main` and PRs targeting `main`, we simply check against the most recent
-  # stable tag.
-  #
-  # For PRs targeting release branches, we should really check against
-  # all the most recent stable tags reachable from either the current branch or
-  # from previous release branches (say, against both `1.17.1` and `1.16.2`
-  # created after the `release/1.17.x` branch).
-  # Instead, we approximate by checking only against the most recent stable tag
-  # reachable from the current branch, under the assumption that if a lesser
-  # release branch contains a protobuf change, then it will also be present in
-  # higher ones either through a shared commit or a back-port from `main`.
```

Comment:
Please don't remove this explanation.

Reply:
I think this adds little value. 

>    For `main` and PRs targeting `main`, we simply check against the most recent stable tag.

This actually describes the ci jobs that we have, so I would much rather add a comment referencing the job definition directly, and add any extra explanations to the job definitions themselves. If not, this will be out of sync sooner or later, as the jobs can change without anyone updating this comment.

>    For PRs targeting release branches, we should really check against
   all the most recent stable tags reachable from either the current branch or
   from previous release branches (say, against both `1.17.1` and `1.16.2`
   created after the `release/1.17.x` branch).
   
This sounds like a TODO which we either track in a ticket and reference the ticket number if we want to do that change, or not add this comment as we will not be doing the change so it makes no sense in mentioning it. Also, my understanding is that checking against older versions than the latest stable doesn't really make sense anyway, as it's transitively compatible. 

>    Instead, we approximate by checking only against the most recent stable tag
   reachable from the current branch, under the assumption that if a lesser
   release branch contains a protobuf change, then it will also be present in
   higher ones either through a shared commit or a back-port from `main`.
   
I would simplify this section and keep it as an explanation for the stable check:

`Check against the most recent stable (non-snapshot) tag, under the assumption that any protobuf changes which are subsequently added to prior releases are also included in the latest stable release tag.`

Reply:
@fabiotudone-da I now understand why that is not really a TODO. But for me, it sounds as though it tries to convey the same meaning as the last message (specifically the second part of the message, which clarifies our assumption that any protobuf changes will be ported to the latest stable tag). 

```
Cherck against the most recent stable (non-snapshot) tag reachable from
the current branch, under the assumption that if a lesser release branch
contains a protobuf change, then it will also be present in higher ones.
```

This sounds good to me, the only thing I would like to double-check is the wording using "lesser release branch" and "higher ones" (I haven't really encountered these quantifiers before when mentioning releases). I assume we mean "previous release branches" and "following releases". Do you think the lesser/higher notation is clearer?

Reply:
I avoided using "previous" and "following" because they tend to imply time rather than version ordering but I guess it's understandable anyway.